content/tutorials/pusher-adaptor.textile

```diff
@@ -0,0 +1,206 @@
+
+  ```[javascript]
+    /* Subscribe to the 'some_channel' channel. This is actually an
+     * Ably channel called 'public:some_channel' */
+    var channel = pusher.subscribe('some_channel');
+    channel.bind('message_name', function(data) {
+      alert("Received a message: " + JSON.stringify(data));
+    });
+
+    /* Publish a message with the Ably client */
+    var ablyChannel = ably.channels.get('public:some_channel');
+    ablyChannel.publish('message_name', { 'some': ['JSON', 'data'] });
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/07671a2
+
+blang[nodejs].
+  Here's some code that subscribes to a channel from the Ably node client library, then publishes to it from the Pusher node client library:
+
+  ```[javascript]
+    /* Subscribe to the 'public:some_channel' channel with the Ably client */
+    const ablyChannel = ably.channels.get('public:some_channel');
+    ablyChannel.subscribe(function(message) {
+      console.log('Ably client received a message: ' + message.name + ', data: ' + JSON.stringify(message.data));
+    });
+
+    /* Publish to the 'some_channel' channel with the Pusher client.
+     * This is a public channel, so the real channel name is 'public:some_channel' */
+    pusher.trigger('some_channel', 'eventName', { 'Some': 'JSON data' });
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/7105d2b
+
+blang[ruby].
+  Here's some code that subscribes to a channel from the Ably ruby client library, then publishes to it from the Pusher ruby client library:
+
+  ```[ruby]
+    # Subscribe to the 'public:some_channel' channel with the Ably client
+    ablyChannel = ably.channels.get('public:some_channel')
+    ablyChannel.subscribe do |message|
+      puts "Ably client received a message: #{message.name} - #{message.data.inspect}"
+    end
+
+    # Publish to the 'some_channel' channel with the Pusher client.
+    # This is a public channel, so the real channel name is 'public:some_channel'
+    pusher.trigger('some_channel', 'event_name', {some: 'data sent by the Pusher client library'})
+  ```
+
+  "See this step in Github":https://github.com/ably/tutorials/commit/fa24b73
+
+h2. Download tutorial source code
+
+blang[javascript].
+  The complete source code of a working example of Ably and Pusher client libraries communicating with each other "is available on Github":https://github.com/ably/tutorials/commits/pusher-adaptor-javascript.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout pusher-adaptor-javascript
+
+  And then run the demo locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @example.html@ and opening the page in your browser.
```

Comment:
Sure, could mention it - was assuming that they'd been following the tutorial, in which case they'd have done that step